### PR TITLE
NOISSUE - Update dependencies

### PR DESCRIPTION
**Pull request by dborovcanin**
# What type of PR is this?

This pull request updates dependencies.
## What does this do?

It updates all dependencies to the latest available versions except (due to backward incompatibility):
- OPC-UA client library in mod file
- Docker version in mod file
- Cassandra version in compose file

## Which issue(s) does this PR fix/relate to?

There is no such issue.

## Have you included tests for your changes?

N/A
## Did you document any new/modified feature?

N/A
### Notes

N/A

### CODE REVIEW

File: readers/mongodb/setup_test.go [hunk 1]
```diff
@@ -23,13 +25,18 @@ func TestMain(m *testing.M) {
...
 	}
 
-	cfg := []string{
-		"MONGO_INITDB_DATABASE=test",
-	}
-
-	container, err := pool.Run("mongo", "4.4.3-bionic", cfg)
+	container, err := pool.RunWithOptions(&dockertest.RunOptions{
+		Repository: "mongo",
+		Tag:        "6.0.13",
```
Comment (rodneyosodo) [inline readers/mongodb/setup_test.go:30]: There is a newer version
```suggestion
		Tag:        "7.0.5",
```

File: readers/cassandra/setup_test.go [hunk 2]
```diff
@@ -22,9 +24,15 @@ func TestMain(m *testing.M) {
 		logger.Error(fmt.Sprintf("Could not connect to docker: %s", err))
 	}
 
-	container, err := pool.Run("cassandra", "3.11.10", []string{})
+	container, err := pool.RunWithOptions(&dockertest.RunOptions{
+		Repository: "cassandra",
+		Tag:        "3.11.16",
```
Comment (rodneyosodo) [inline readers/cassandra/setup_test.go:29]: Do we support version 5.0?

File: docker/docker-compose.yml [hunk 3]
```diff
@@ -58,7 +58,7 @@ services:
       - spicedb-db
 
   spicedb-db:
-    image: "postgres:15.3-alpine"
+    image: "postgres:16.1-alpine"
```
Comment (rodneyosodo) [inline docker/docker-compose.yml:61]: Also update jaeger, redis, nginx, users and invitations db

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

File: readers/cassandra/setup_test.go (hunk 2 above)
Comment (dborovcanin) [reply readers/cassandra/setup_test.go:29]: At the moment, both 4.x.x and 5.0 are not supported.

**Review by rodneyosodo — CHANGES_REQUESTED** [review]
The following cases have not been updated
- `users/postgres/setup_test.go` 
- `things/postgres/setup_test.go`
- `invitations/postgres/setup_test.go`
- `internal/groups/postgres/setup_test.go`
- `pkg/clients/postgres/setup_test.go`
- `pkg/messaging/mqtt/setup_test.go`
- `pkg/messaging/nats/setup_test.go` and `pkg/events/nats/setup_test.go`
- `pkg/messaging/rabbitmq/setup_test.go` and `pkg/events/rabbitmq/setup_test.go`
- `pkg/events/redis/setup_test.go` and `bootstrap/events/producer/setup_test.go`
- `auth/postgres/setup_test.go`
- `bootstrap/postgres/setup_test.go`
- `certs/postgres/setup_test.go`
- `consumers/notifiers/postgres/setup_test.go`
- Redis on `docker/addons/lora-adapter/docker-compose.yml`, `docker/addons/opcua-adapter/docker-compose.yml` , `docker/addons/twins/docker-compose.yml` and `docker/docker-compose.yml`
- `docker/addons/prometheus/docker-compose.yml`
- `docker/addons/smpp-notifier/docker-compose.yml`
- `docker/addons/smtp-notifier/docker-compose.yml`
- `docker/addons/vault/docker-compose.yml`
- `docker/es` and `docker/brokers`

File: docker/addons/vault/docker-compose.yml [hunk 4]
```diff
@@ -18,7 +18,7 @@ volumes:
 
 services:
   vault:
-    image: vault:1.12.2
+    image: vault:1.15.4
```
Comment (rodneyosodo) [inline docker/addons/vault/docker-compose.yml]:
```suggestion
    image: hashicorp/vault:1.15.4
```

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

**Review by rodneyosodo — APPROVED** [review]

**Pull request merged**